File: leetcode/rust/number-of-steps-to-reduce-a-number-to-zero.rs
impl Solution {
    pub fn number_of_steps(num: i32) -> i32 {
        let mut num = num;
        let mut result = 0;

        while num > 0 {
            num = if num % 2 == 0 { num / 2 } else { num - 1 };

            result += 1;
        }

        return result;
    }
}

File: leetcode/js/number-of-steps-to-reduce-a-number-to-zero.js
/**
 * @param {number} num
 * @return {number}
 */
var numberOfSteps = function (num) {
  let result = 0;

  while (num > 0) {
    num = num % 2 === 0 ? num / 2 : num - 1;
    result++;
  }

  return result;
};
